url2698a2c731_cassandra_project_cassandra_stand_alone_tgz-pJ8-eu_cassandra_sim_MySimpleDynamicScenarioSimulationJ8/src/eu/cassandra/sim/StandAloneSimulation.java

```diff
@@ -0,0 +1,527 @@
+  			double offpeakPrice = Double.parseDouble((sfp.propPricing.getProperty("offpeakPrice") != null ? sfp.propPricing.getProperty("offpeakPrice").trim() : "0"));
+  			String[] levelsAS = sfp.propPricing.getProperty("levels").replace("[", "").replace("]", "").split(",");
+  			String[] pricesAS = sfp.propPricing.getProperty("prices").replace("[", "").replace("]", "").split(","); 
+  			if (levelsAS.length != pricesAS.length)
+  				throw new Exception("ERROR: levels and prices lists must have the same length");
+  			double[] levelsA = new double[levelsAS.length];
+  			double[] pricesA = new double[levelsAS.length];
+  			for(int i = 0; i < levelsAS.length; i++) {
+  				levelsA[i] = Double.parseDouble(levelsAS[i]);
+  				pricesA[i] = Double.parseDouble(pricesAS[i]);
+  			}
+  			String[] offpeakHours = sfp.propPricing.getProperty("offpeakHours").replace("[", "").replace("]", "").split(",");
+  			if (offpeakHours.length == 1 && offpeakHours[0].equals(""))
+  				offpeakHours = new String[0];
+  			String[] fromsA = new String[offpeakHours.length];
+  			String[] tosA = new String[offpeakHours.length];
+  			for(int i = 0; i < offpeakHours.length; i++) {
+  				String from = offpeakHours[i].split("-")[0].trim();
+  				String to = offpeakHours[i].split("-")[1].trim();
+  				fromsA[i] = from;
+  				tosA[i] = to;
+  			}
+  			builderPP.scalarEnergyPricingTimeZones(offpeakPrice, pricesA, levelsA, fromsA, tosA);
+  			break;
+  		case "EnergyPowerPricing":
+  			double contractedCapacity = Integer.parseInt((sfp.propPricing.getProperty("contractedCapacity") != null ? sfp.propPricing.getProperty("contractedCapacity").trim() : "0"));
+  			double energyPricing = Double.parseDouble((sfp.propPricing.getProperty("energyPrice") != null ? sfp.propPricing.getProperty("energyPrice").trim() : "0.0"));
+  			double powerPricing = Double.parseDouble((sfp.propPricing.getProperty("powerPrice") != null ? sfp.propPricing.getProperty("powerPrice").trim() : "0.0"));
+  			builderPP.energyPowerPricing(contractedCapacity, energyPricing, powerPricing);
+  			break;
+  		case "AllInclusivePricing":
+  			int fixedCost = Integer.parseInt((sfp.propPricing.getProperty("fixedCost") != null ? sfp.propPricing.getProperty("fixedCost").trim() : "0"));
+  			double additionalCost = Double.parseDouble((sfp.propPricing.getProperty("additionalCost") != null ? sfp.propPricing.getProperty("additionalCost").trim() : "0.0"));
+  			double contractedEnergy = Double.parseDouble((sfp.propPricing.getProperty("contractedEnergy") != null ? sfp.propPricing.getProperty("contractedEnergy").trim() : "0.0"));
+  			builderPP.allInclusivePricing(fixedCost, additionalCost, contractedEnergy);
+  			break;
+  		default:
+			throw new Exception("ERROR: unkown pricing policy type \"" + pricingType + "\" employed");
+  		}
+		return builderPP.build();
+	}
+	
+	private ProbabilityDistribution constructDistribution(String distType, Properties prop, String caseD) throws Exception
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 32. By comparison, 99% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test. We recommend that you simplify this method or break it into multiple methods.